authorization_services/topics/enforcer-claim-information-point.adoc

```diff
@@ -0,0 +1,165 @@
+[[_enforcer_claim_information_point]]
+= Claim Information Point
+
+A Claim Information Point (CIP) is responsible for resolving claims and pushing these claims to the {project_name} server in order
+to provide to policies more information about the access context. They can be defined as a configuration option to the `policy-enforcer`
+in order to resolve claims from different sources, such as:
+
+* HTTP Request (parameters, headers, body, headers, etc)
+* External HTTP Service
+* Static values defined in configuration
+* Any other source by implementing the Claim Information Provider SPI
+
+When pushing claims to the {project_name} server, policies can base their decisions not just based on who an user is but taking context
+and contents into account, based upon who, what, why, when, where and which for a given transaction. It is all about Contextual-based Authorization
+and how to use runtime information in order to support fine-grained authorization decisions.
+
+== Obtaining information from the HTTP Request
+
+Here are several examples about how you can extract claims from an HTTP request:
+
+.keycloak.json
+```json
+"policy-enforcer": {
+    "paths": [
+      {
+        "path": "/protected/resource",
+        "claim-information-point": {
+          "claims": {
+            "claim-from-request-parameter": "{request.parameter['a']}",
+            "claim-from-header": "{request.header['b']}",
+            "claim-from-cookie": "{request.cookie['c']}",
+            "claim-from-remoteAddr": "{request.remoteAddr}",
+            "claim-from-method": "{request.method}",
+            "claim-from-uri": "{request.uri}",
+            "claim-from-relativePath": "{request.relativePath}",
+            "claim-from-secure": "{request.secure}",
+            "claim-from-json-body-object": "{request.body['/a/b/c']}",
+            "claim-from-json-body-array": "{request.body['/d/1']}",
+            "claim-from-body": "{request.body}",
+            "claim-from-static-value": "static value",
+            "claim-from-multiple-static-value": ["static", "value"],
+            "param-replace-multiple-placeholder": "Test {keycloak.access_token['/custom_claim/0']} and {request.parameter['a']} "
+          }
+        }
+      }
+    ]
+  }
+```
+
+== Obtaining information from an External HTTP Service
+
+Here are several examples about how you can extract claims from an external HTTP Service:
+
```

Comment:
Here are several examples showing how you can extract claims from an external HTTP Service: